conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/ConjureFeignJacksonEncoder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.jaxrs;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.RequestTemplate;
+import feign.codec.EncodeException;
+import feign.codec.Encoder;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Similar to {@link feign.jackson.JacksonEncoder}, but optimized to avoid intermediate
+ * String representation of request body.
+ * See upstream PR https://github.com/OpenFeign/feign/pull/989 .
+ */
+final class ConjureFeignJacksonEncoder implements Encoder {
+
+    private final ObjectMapper mapper;
+
+    ConjureFeignJacksonEncoder(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public void encode(Object object, Type bodyType, RequestTemplate template) {
+        try {
+            JavaType javaType = mapper.getTypeFactory().constructType(bodyType);
+            template.body(mapper.writerFor(javaType).writeValueAsBytes(object), StandardCharsets.UTF_8);
```

Comment:
Eyeballing this, I wondered whether the plain `mapper.writeValueAsBytes()` might avoid allocating the objectwriter, but I guess the reflection is more costly.

This is also consistent with what the upstream feign jackson encoder used to do:

```java
template.body(mapper.writerFor(javaType).writeValueAsString(object));
```

Reply:
This implementation bounds the writer to the type specified on our interface, rather than Object. It's unlikely that it makes a difference for our use in conjure, but it does provide additional specificity.